status/data_deliveries_plot.py

```diff
@@ -0,0 +1,40 @@
+import json
+import datetime
+from status.util import SafeHandler
+
+
+class DataDeliveryHandler(SafeHandler):
+    """ Handles the api call to proj_staged data
+    Loaded through /api/v1/proj_staged/([^/]*)$
+    """
+
+    def get(self, search_string=None):
+        staged_files_sum_view = self.application.projects_db.view("project/staged_files_sum")
+
+        today = datetime.date.today()
+        if search_string:
+            start_date, end_date = search_string.split('--')
+        else:
+            last_month = today - datetime.timedelta(days=30)
+            start_date = last_month.isoformat()
+            end_date = today.isoformat()
+
+        docs = staged_files_sum_view.rows
+        # Projects without close date are filtered out
+        pids = [d.key for d in docs if start_date <= d.value.get('close_date', 'ZZZZ-ZZ-ZZ') < end_date]
+        vals = [d.value for d in docs if start_date <= d.value.get('close_date', 'ZZZZ-ZZ-ZZ') < end_date]
+        data = {k:v for (k,v) in zip(pids, vals)} 
```

Comment:
I think you could replace the three lines with:
```
data = [(d.key, d.value) for d in docs if start_date <= d.value.get('close_date', 'ZZZZ-ZZ-ZZ') < end_date]
```
and have the same result?

Reply:
Yes! I tried to do it in a one-liner, but didn't manage, thanks!